improver_tests/calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py

```diff
@@ -475,36 +408,31 @@ def test_end_to_end(self):
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
     def test_end_to_end_with_mask(self):
-        """An example end-to-end calculation, this time using a mask to limit
-        the application of calibration to regions where the mask=1."""
+        """An example end-to-end calculation, but making sure that the
+        areas that are masked within the landsea mask, are masked at the
+        end."""
 
         # Construct a mask and encapsulate as a cube.
-        mask = np.ones((3, 3))
-        mask[1:, 1:] = 0
-        mask_cube = self.current_temperature_forecast_cube[0].copy(data=mask)
-
-        # Collapse input realizations to get uncalibrated expectation values.
-        forecast_predictors = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.MEAN)
-        forecast_vars = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-
-        # Manually construct merged calibrated and uncalibrated arrays.
-        self.expected_loc_param_mean[1:, 1:] = (
-            forecast_predictors.data[1:, 1:])
-        self.expected_scale_param_mean[1:, 1:] = (
-            forecast_vars.data[1:, 1:])
+        mask = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+        mask_cube = self.current_temperature_forecast_cube[0].copy(
+            data=mask)
+        predictor = np.array([[273.78543, 274.69128, 275.4461],
+                              [276.86517, 277.65018, 278.40503],
+                              [279.492, 280.15625, 280.97147]])
+        mask_predictor = np.ma.masked_where(mask, predictor)
```

Comment:
I think that `predictor` is the same as `self.expected_loc_param_mean`, so this could just be:
```
mask_predictor = np.ma.masked_where(mask, self.expected_loc_param_mean)
```